### ISSUE TITLE
Progress indication for non-terminals

### ISSUE THREAD

**Issue by nponeccop** [ndmitchell/hoogle#231]
Currently `hoogle generate` looks pretty ugly if it writes to a file and/or the terminal is broken. For example it's how it looks on CircleCI build log if `hoogle generate` is executed during a build: 

```
~/.local/bin/hoogle generate
Starting generate
Reading Cabal... 9.79s
[1/2523] AC-Vector... 0.04s                           [2/2523] IfElse... 0.00s                        [3/2523] Imlib... 0.04s                       [4/2523] NineP... 0.02s                       [5/2523] OneTuple... 0.06s                          [6/2523] PSQueue... 0.01s                         [7/2523] ParsecTools... 0.00s                             [8/2523] SegmentTree... 0.00s                             [9/2523] X11-xft... 0.01s                         [10/2523] Xauth... 0.00
```
Visually it's like this:

![image](https://user-images.githubusercontent.com/528415/33800267-c88ee2f6-dd12-11e7-82c3-6dfed7559b17.png)

My suggestion is to have a simpler progress indicator designed for logs and incorrect terminal emulators. Something along the lines of `--no-terminal` option of `stack`.

**Comment by ndmitchell** [comment]
Sounds sensible - I'd happily accept a PR. One option would be using the `hSupportsANSI` from ansi-terminal or `isTerminal` from base (I'm happy either way).

**Comment by 9b17fe** [comment]
This seems to have been fixed since 5245f28baa12638b06c660c34df66867e88f892c.

**Comment by ndmitchell** [comment]
Oh, awesome! Closing optimistically.
